File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> sta;
        for(int i =0;i<s.length();i++){
            if(s[i]=='(' || s[i]=='[' || s[i]=='{'){
                sta.push(s[i]);
            }
            else if(s[i]==')' || s[i]==']' || s[i]=='}'){
                if(sta.empty()){
                    return false;
                }
                else{
                    char temp = sta.top();
                    sta.pop();
                    if(!((temp == '(' && s[i] == ')') || (temp == '{' && s[i] == '}') || (temp == '[' && s[i] == ']'))){
                        return false;
                    }
                }
            }
        }
        if(sta.empty()){
             return true;
        }
       return false;
    }
};

File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<Character>();
        for(int i =0;i<s.length();i++){
            if(s.charAt(i)=='('||s.charAt(i)=='['||s.charAt(i)=='{'){
                st.push((s.charAt(i)));
            }
            else if(s.charAt(i)==')'||s.charAt(i)==']'||s.charAt(i)=='}'){
                if(st.isEmpty()){
                    return false;
                }
                else{
                    char c = st.peek();
                    st.pop();
                    if(!((s.charAt(i)==')' && c=='(')||(s.charAt(i)==']' && c=='[')||(s.charAt(i)=='}'&& c=='{'))){
                        return false;
                    }
                }
            }
        }
        if(st.isEmpty()){
            return true;
        }
        return false;
    }
}